### PR TITLE
Support for kopeio-networking

### PR DESCRIPTION
**Pull request by justinsb**
Also cherry-pick configuration-through-code out of the code where it was (authn)

### CODE REVIEW

**Comment by chrislovecnm** [comment]
You need to name you routing package, please do not use vxlan, lmao

File: channels/pkg/channels/channel_version.go [hunk 1]
```diff
@@ -39,6 +40,17 @@ type ChannelVersion struct {
...
 }
 
+func stringValue(s *string) string {
+	if s == nil {
+		return ""
+	}
+	return *s
+}
+
+func (c *ChannelVersion) String() {
```
Comment (JuJu227) [inline channels/pkg/channels/channel_version.go]: @justinsb out of curiosity, what's the need for reference/pointer string `*string` inside of a struct?  By default behavior of a string, the initial value is a blank string, not a nil and you wouldn't need a helper function to do what's already a functionality of the language. The *ChannelVersion is already a copy of the address block you allocated. I'm trying to wrap my head around things.

Comment (justinsb) [inline channels/pkg/channels/channel_version.go]: It allows for a difference between not-set and set-to-empty.  It avoids the need for per-field logic to know whether empty means empty or not-set.

**Comment by chrislovecnm** [comment]
We need to have private topology allow for vxlan and cni.  Right now it is coded to only allow cni.  https://github.com/justinsb/kops/blob/8bec70b1388fc2b5c3a2603254057f3908c4f6f1/cmd/kops/create_cluster.go#L375

**Comment by chrislovecnm** [comment]
Also please update the name `vxlan` to something else :)

**Comment by justinsb** [comment]
Called `--networking kopeio-vxlan`.  Represented in the data model as "networking: kopeio: {}"

**Pull request merged**